rest/data/project.go

```diff
@@ -86,6 +86,18 @@ func (ac *DBProjectConnector) GetProjectEventLog(id string, before time.Time, n
 	return out, catcher.Resolve()
 }
 
+func (ac *DBProjectConnector) GetProjectWithCommitQByOwnerRepoAndBranch(owner, repo, branch string) (*model.ProjectRef, error) {
+	proj, err := model.FindOneProjectRefWithCommitQByOwnerRepoAndBranch(owner, repo, branch)
```

Comment:
this seems like a new function, could we just return the error when there weren't results, rather than doing the nil/nil thing?

Reply:
done.